src/libs/IntlPolyfill/index.ios.ts

```diff
@@ -7,12 +7,21 @@ import type IntlPolyfill from './types';
  * Polyfill the Intl API, always performed for native devices.
  */
 const intlPolyfill: IntlPolyfill = () => {
-    // Native devices require extra polyfills
-    require('@formatjs/intl-getcanonicallocales/polyfill');
+    // Native devices require extra polyfills which are
+    // not yet implemented in hermes.
+    // see support: https://hermesengine.dev/docs/intl/
+
     require('@formatjs/intl-locale/polyfill');
+
+    // Required to polyfill NumberFormat on iOS
+    // see: https://github.com/facebook/hermes/issues/1172#issuecomment-1776156538
     require('@formatjs/intl-pluralrules/polyfill');
     polyfillNumberFormat();
```

Comment:
There is an additional check inside that function: `hasOldCurrencyData`. Seems like even if the API is supported it may not work as expected e.g. old API. Seeing that we have an early return in enabling that polyfill it may be safer to keep it

Reply:
@s77rt  I have some reservations around `hasOldCurrencyData`, if you could explain a bit more about it. So the thing is the output of this function differs only by a space character, otherwise both of the strings are the same on all platforms. 

I printed both values of the comparison in the console and it appears that the output by `NumberFormat` contains a special space character which is different from normal space character.

<img width="1188" alt="Screenshot 2024-03-21 at 1 33 02 PM" src="https://github.com/Expensify/App/assets/47336142/4c79a003-b3ec-4955-9e68-73e27addb1dd">

If I update this const value, only then we do early return otherwise we are always polyfilling `NumberFormat`. Android and Web also works fine BUT iOS doesn't. Which means that the below code is not working as expected. Ideally, iOS should not fall in the early-return. We probably want to update this condition to include `Intl.NumberFormat(locale, options).formatToParts(number)` because this is what's missing in iOS and it causes the crash. 



```ts

/**
 * Check if the locale data is as expected on the device.
 * Ensures that the currency data is consistent across devices.
 */
function hasOldCurrencyData(): boolean {
    return (
        new Intl.NumberFormat(CONST.LOCALES.DEFAULT, {
            style: CONST.POLYFILL_TEST.STYLE,
            currency: CONST.POLYFILL_TEST.CURRENCY,
            currencyDisplay: CONST.POLYFILL_TEST.FORMAT,
        }).format(Number(CONST.POLYFILL_TEST.SAMPLE_INPUT)) !== CONST.POLYFILL_TEST.EXPECTED_OUTPUT
    );
}

export default function () {
    if (Intl && 'NumberFormat' in Intl && !hasOldCurrencyData()) {
        return;
    }

    require('@formatjs/intl-numberformat/polyfill-force');

    // Load en & es Locale data
    require('@formatjs/intl-numberformat/locale-data/en');
    require('@formatjs/intl-numberformat/locale-data/es');
}


```

Reply:
iOS do not fall into the early-return, perhaps you tested after the polyfill was applied already.
<img width="308" alt="Screenshot 2024-03-22 at 12 10 22 AM" src="https://github.com/Expensify/App/assets/16493223/000a6251-9c42-4ba8-95e4-f2002caeed56">


Got same results on Android

Reply:
@s77rt Did you test with the applied diff? I also get the same results as you but without the diff which is what we have in main.

With the diff, each platform falls into the early return and only iOS fails because polyfill isn't applied and we get error on `Intl.NumberFormat(locale, options).formatToParts(number)`. So I am wondering whether we should include `formatToParts` in the condition for early return?

Otherwise, let's keep the things as is 👍

Reply:
@hurali97 Let's do the following:
1. Fix the `POLYFILL_TEST.EXPECTED_OUTPUT` value (use correct whitespace) so the polyfill is applied only when needed
2. Add `formatToParts` to the condition

Summary:
- Intl.NumberFormat polyfill is not needed on Android.
- Intl.NumberFormat polyfill is needed on iOS due to missing formatToParts funcition
- Intl.NumberFormat polyfill is needed on Web due to outdated currency data on some browsers ([issue](https://github.com/Expensify/App/issues/14307))

Reply:
@s77rt Perfect. I have addressed these changes, you can re-review 👍